casper/src/test/scala/coop/rchain/casper/api/PreviewPrivateNameTest.scala

```diff
@@ -0,0 +1,58 @@
+package coop.rchain.casper.api
+
+import cats.Id
+import com.google.protobuf.{ByteString, Int64Value}
+import coop.rchain.casper.protocol.DeployData
+import coop.rchain.casper.util.ProtoUtil
+import coop.rchain.crypto.codec.Base16
+import coop.rchain.crypto.hash.Blake2b512Random
+import coop.rchain.models.{GPrivate}
+import coop.rchain.p2p.EffectsTestInstances.LogStub
+
+import org.scalatest.{FlatSpec, Matchers}
+
+class PreviewPrivateNameTest extends FlatSpec with Matchers {
+  implicit val logEff = new LogStub[Id]
+
+  def previewId(pkHex: String, timestamp: Long, nth: Int = 0): String = {
+    val preview = BlockAPI.previewPrivateNames[Id](
+      ProtoUtil.stringToByteString(pkHex),
+      timestamp,
+      nth + 1
+    )
+
+    Base16.encode(preview.ids(nth).toByteArray)
+  }
+
+  val myNodePk = "464f6780d71b724525be14348b59c53dc8795346dfd7576c9f01c397ee7523e6"
+
+  "previewPrivateNames" should "work in one case" in {
+    // When we deploy `new x ...` code from a javascript gRPC client,
+    // we get this private name id in the log:
+    // 16:41:08.995 [node-runner-15] INFO  c.r.casper.MultiParentCasperImpl - Received Deploy #1542308065454 -- new x0, x1 in {
+    //   @{x1}!(...
+    // [Unforgeable(0xb5630d1bfb836635126ee7f2770873937933679e38146b1ddfbfcc14d7d8a787), bundle+ {   Unforgeable(0x00) }]
+    // 2018-11-15T18:54:25.454Z
+    previewId(myNodePk, 1542308065454L) should equal(
+      "b5630d1bfb836635126ee7f2770873937933679e38146b1ddfbfcc14d7d8a787"
+    )
+  }
```

Comment:
Does this work for predicting `x2` in

`new x0, x1 in { x0!(5) | for (_ <- x0) { new x2 in { ... } } }`?

Reply:
I doubt it.

I suppose that should be documented better.

Meanwhile, I'll test to see if I get lucky...

Reply:
Nope. The preview gRPC gives:

```
[ '64b2d2e4dcbb7a0d691431ea0aa270f88f3cfa19bd4740cfced007e578bc98a3',
  'de727884e43d52a7b8b9c532f4e6634212243fd2567fa4c6e1709a48051deef9',
  '57ec7b8bd6491bf0dbf31b0de2c4e850dc5a07941ba50b8ab5362c412ff30749' ]
```

but when I put `stdout!([*x0, *x1, *x2])` in for `...` in the above, I get:
```
[Unforgeable(0x64b2d2e4dcbb7a0d691431ea0aa270f88f3cfa19bd4740cfced007e578bc98a3),
 Unforgeable(0xde727884e43d52a7b8b9c532f4e6634212243fd2567fa4c6e1709a48051deef9),
 Unforgeable(0x0ab721c6da23a6d2ad1d62cd1fea661cc860f6e3d5cc7aa042d442109b5d109f)]
```

Reply:
It won't work because the seed is being split for `send`s and then merged for `consume`s.

Reply:
right... figuring out where the splits and merges happen is basically equivalent to writing a rholang interpreter